### PR TITLE
MC: Takeoff and land support

### LINKED ISSUE #3365: MC: Add auto land support analog to auto takeoff

### PR DESCRIPTION
**Pull request by AndreasAntener**
Merry Christmas PX4 :)
Adding with permission, takeoff and landing improvements.

This PR adds:
- Auto takeoff
- Auto land
- ACK for commands
- Possibility to switch to POSCTL/ALTCTL on the ground (to really make use of this #3385 needs to be implemented)
- Make RTL use the same landing method

Disarming after landing is not active by default, we'd need to check default values for that carefully. How good it works depends on the tuning of the airframe, all the involved parameters have to be in "sync".

Landing and takeoff is currently supported via command, in missions and RTL. #3385 could be the next step.

Fixes #3365

### CODE REVIEW

**Comment by mhkabir** [comment]
@AndreasAntener Looking great! I'd love to give this more testing coverage. 

Could you tell me how the commands are accepted and handled?

**Comment by AndreasAntener** [comment]
Hey @mhkabir 
Sure, testing is always appreciated :). The commanded takeoff and landing are actually implemented as flight modes. So you can just put it into auto takeoff or auto land mode and it will do it. Internally the position setpoints then have the types takeoff or land. So if you're doing something more direct with setpoints, setting the setpoint type to takeoff/land will result in the same behaviour.

**Comment by AndreasAntener** [comment]
In case you remember what we talked about regarding offboard and setpoint type: https://github.com/UAVenture/Firmware/commit/0e6ccf1ed9557516543e100bb9bceee9d801c2ee
Been using this for a while now but haven't gotten around yet to put this in a PR. Will followup with that soon.

**Comment by mhkabir** [comment]
Great! Yes, I was just about to follow-up on our prior discussion, but it looks like you already have all the bases covered :)

> So you can just put it into auto takeoff or auto land mode and it will do it

So I assume that only on receiving the command, it will switch to auto takeoff/ auto land? What happens when it reaches target altitude? What mode does it switch to? Is there any prerequisite to get the command accepted (e.g be in any particular mode, or be armed, etc) ?
Also, how are the transmitter mode settings handled? Can these auto takeoff/land modes be switched to from a transmitter?

File: src/lib/conversion/rotation.h [hunk 1]
```diff
@@ -117,7 +118,7 @@ const rot_lookup_t rot_lookup[] = {
 	{270,   0, 270 },
 	{180, 270,   0 },
 	{  0,  90, 180 },
-	{ 90,  90,   0 }
```
Comment (thomasgubler) [inline src/lib/conversion/rotation.h]: are you sure about the removal of this entry?

Comment (AndreasAntener) [inline src/lib/conversion/rotation.h]: yes I'm sure the removal is wrong ;) thanks. must have slipped during the merge conflict

**Comment by AndreasAntener** [comment]
@mhkabir no switching with transmitter yet. #3385 would be cool

It needs GPS for those modes, landing should also be possible without (haven't tried though). After takeoff/land it's going to stay in that mode until you command something else, except if you were in ALT/POSCTL before it's going to switch back to that.

**Comment by LorenzMeier** [comment]
@AndreasAntener What's going on with the build failure here? Maybe a rebase will fix that? I improved flash usage recently.

**Comment by AndreasAntener** [comment]
@LorenzMeier rebased. But I also get the flash error on latest master using `gcc-arm-none-eabi-48/20140805`

**Comment by LorenzMeier** [comment]
Apparently it passed. You might need to clean

**Comment by LorenzMeier** [comment]
Applied to master, thanks!

**Pull request closed without merging**